packages/core/botpress/src/util.js

```diff
@@ -123,6 +127,50 @@ const getPackageName = pkg => {
   }
 }
 
+const versionExists = version => {
+  if (version == null) {
+    throw new Error("Version doesn't exist in botfile.js")
+  }
+
+  return version
+}
+const isValidString = version => {
+  if (_.isEmpty(version) || !_.isString(version)) {
+    throw new Error('Version must be non-empty string')
+  }
+
+  return version
+}
+const isValidFormat = version => {
+  try {
+    // TODO: change this method if "semver" module will implement semver.isValid()
+    semver.valid(version)
+  } catch (err) {
+    throw new Error('Version must have semver format (e.g. 10.25.0)')
+  }
+
+  return version
+}
+const isEqualOrGreater = packageVersion => botfileVersion => {
+  const botfileMajorVersion = Number(semver.major(botfileVersion))
+  const packageMajorVersion = Number(semver.major(packageVersion))
+
+  const msg = `Your bot may be incompatible with botpress v${packageMajorVersion}.XX.x
+  because it looks like your bot was originally created with botpress v${botfileMajorVersion}.XX.x.
+  To address this:
+
+  - please change the version property in your botfile to be "${packageMajorVersion}"`
```

Comment:
just changing the version is pointless, the actions depend on the specific condition:
1) if the major versions are identical the fix is to update to `botpress v${botfileVersion}` by changing the botpress and all `@botpress/*` modules version in the _bot's_ package.json
2) if the botpress major version is below the botfile's major version — same
3) if the botpress major version is above the botfile's major version - link to CHANGELOG in the repo root and suggest looking for and addressing the breaking changes listed there

Reply:
@emirotin done